torchsde/_core/methods/heun.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ...settings import SDE_TYPES, NOISE_TYPES, LEVY_AREA_APPROXIMATIONS
+
+from .. import base_solver
+
+
+class Heun(base_solver.BaseSDESolver):
+    weak_order = 1.0
+    sde_type = SDE_TYPES.stratonovich
+    noise_types = (NOISE_TYPES.additive, NOISE_TYPES.diagonal, NOISE_TYPES.general, NOISE_TYPES.scalar)
+    levy_area_approximations = LEVY_AREA_APPROXIMATIONS.all()
+
+    def __init__(self, sde, **kwargs):
+        if sde.noise_type == NOISE_TYPES.general:
+            self.strong_order = 0.5
```

Comment:
I think I only forgot update it: both should be 1.0?

Reply:
Heun is 0.5 for general noise an 1.0 for commutative (diagonal/scalar/additive) noise, so this is correct as-is.